src/PerunDummy.hs

```diff
@@ -556,12 +691,20 @@ findChannel cID = do
         Just d@ChannelDatum {} -> return (oref, o, d)
```

Comment:
Github does not allow me to comment further up in line `687`.

This `case` expression only works for a list with a **single** entry. If for some reason, there are multiple entries this would `throwError "channel utxo not found".

Reply:
If you want to match any list length but only the first entry then use `a:_` or `a:as` if you need access to the rest of the list. In this case it would be `(oref, o):_` to match any list of at least length `1`.

Reply:
Also you can simplify even further:

```haskell
findChannel ::
  ChannelID ->
  Contract w s Text (TxOutRef, ChainIndexTxOut, ChannelDatum)
findChannel cID = do
  utxos <- utxosAt $ scriptHashAddress (channelHash cID)
  case Map.toList utxos of
    [(oref, o)] -> case _ciTxOutDatum o of
      Left _ -> throwError "datum missing"
      Right (Datum e) -> case PlutusTx.fromBuiltinData e of
        Nothing -> throwError "datum has wrong type"
        Just d@ChannelDatum {} -> return (oref, o, d)
    _ -> throwError "channel utxo not found"
```